### DISCUSSION TITLE
Help! Volcano plot in R with seurat and ggplot

### DISCUSSION THREAD

**Discussion by scRNAseqnovice** [satijalab/seurat#6674] [Q&A]
Hi, I am a novice in analyzing scRNAseq data. I have been following the Satija lab tutorials and have found them intuitive and useful so far.

I would like to create a volcano plot to compare differentially expressed genes (DEGs) across two samples- a "before" and "after" treatment.

I have scoured the web but I still cannot figure out how to do this.

I have successfully installed ggplot, normalized my datasets, merged the datasets, etc., but what I do not understand is how to transfer the sequencing data to the ggplot function. I keep receiving an error that says:

"data must be a <data.frame>, or an object coercible by fortify(), not an S4 object with class ."

I have seen tutorials on the web, but the data there is not processed the same as how I have been doing following the Satija lab method, and, my files are not .csv, but instead are .tsv

Any tips would be appreciated!

Thanks!

**Comment by wvictor14** [comment]
See Seurat's DEG vignette: 

https://satijalab.org/seurat/articles/de_vignette.html

It sounds like you want to compare within a cell cluster, between cells from before and after treatment.

The general process for detecting genes then would be:

1. make sure label exists on your cells in the metadata corresponding to treatment (before- and after-)
2. run `FindMarkers` on your processed data, setting ident.1 and ident.2 to correspond to before- and after- labelled cells
3. You will be returned a gene list of pvalues + logFc + other statistics. With this data you can now make a volcano plot

Repeat for all cell clusters/types of interest, depending on your research questions.

**Reply by scRNAseqnovice** [reply]
thank you very much!
